BungeeCord-Patches/0060-Waterdog-PE-modifications.patch

```diff
@@ -1019,6 +1019,100 @@ index 6ef76ea2..5dee53dd 100644
      }
  
      protected void rewriteV8ChunkSection(ByteBuf in, ByteBuf out) {
+diff --git a/protocol/src/main/java/network/ycc/waterdog/pe/PERawPacketData.java b/protocol/src/main/java/network/ycc/waterdog/pe/PERawPacketData.java
+index 55ca27bd..88074783 100644
+--- a/protocol/src/main/java/network/ycc/waterdog/pe/PERawPacketData.java
++++ b/protocol/src/main/java/network/ycc/waterdog/pe/PERawPacketData.java
+@@ -9,6 +9,7 @@ import io.netty.channel.Channel;
+ import net.md_5.bungee.protocol.DefinedPacket;
+ import net.md_5.bungee.protocol.ProtocolConstants;
+ 
++import java.util.Map;
+ import java.util.UUID;
+ 
+ public class PERawPacketData {
+@@ -173,4 +174,24 @@ public class PERawPacketData {
+         channel.write(removeEntityEffect);
+     }
+ 
++    public static void injectSetGameRule(Channel channel, final Map<String, Object> rules){
```

Comment:
Do we need to do this on the bungee side?

Reply:
Yes. Because game rules are sent in Login Packet. Or only when it is forced to change manually (ex. gamerule command, or by plugin).